specification/search_application/post_behavioral_analytics_event/BehavioralAnalyticsEventPostRequest.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { RequestBase } from '@_types/Base'
+import { Name } from '@_types/common'
+import { EventType } from '../_types/AnalyticsEvent'
+
+/**
+ * Create a behavioral analytics collection event.
+ * @rest_spec_name search_application.post_behavioral_analytics_event
+ * @availability stack stability=experimental visibility=public
+ * @doc_tag analytics
+ * @doc_id behavioral-analytics-collection-event
+ * @ext_doc_id behavioral-analytics-event-reference
+ */
+export interface Request extends RequestBase {
+  path_parts: {
+    /**
+     * The name of the behavioral analytics collection.
+     */
+    collection_name: Name
+    /**
+     * The analytics event type.
+     */
+    event_type: EventType
+  }
+}
```

Comment:
```suggestion
}
  query_parameters: {
    /**
     * Whether the response type has to include more details
     */
    debug?: boolean
  }
  /** @codegen_name payload */
  body: UserDefinedValue
```

the body seems to accept a binary payload on the server side, so I'd go with UserDefinedValue

Reply:
While the server code initially accepts a binary payload, it's later translated to ECS, as documented in https://www.elastic.co/guide/en/elasticsearch/reference/current/post-analytics-collection-event.html and https://www.elastic.co/guide/en/elasticsearch/reference/current/behavioral-analytics-event-reference.html. That said, UserDefinedValue is fine for an initial approach.